src/layers.js

```diff
@@ -1,4 +1,25 @@
-const transformLayers = (layers, transform, predicate = () => true) => {
+// @flow
+import { TypeLayer, TypeLayerPredicate, TypeMatrix } from "./types/types";
+
+const transformLayers = (
+  layers: Array<TypeLayer>,
+  transform:
+    | {
+        w: number,
+        h: number,
+        cx: number,
+        cy: number,
+        relative: boolean
+      }
+    | {
+        cx?: number,
+        cy?: number,
+        x: number,
+        y: number,
+        relative: boolean
+      },
```

Comment:
I don't think this type is quite right. You can pass `w` and `h` along with `x` and `y`, but they're always optional. Similarly, `cx`, `cy` and `relative` are all optional. I think it's more like this:

```js
{
  x?: number,
  y?: number,
  w?: number,
  h?: number,
  cx?: number,
  cy?: number,
  relative?: boolean
}
```